diff --git a/demo.py b/demo.py
index 9df169f..8ef06e7 100644
--- a/demo.py
+++ b/demo.py
@@ -1,4 +1,5 @@
 import sys
+import os
 sys.path.append('core')
 
 import argparse
@@ -46,7 +47,7 @@ def demo(args):
             _, flow_up = model(image1, image2, iters=args.valid_iters, test_mode=True)
             flow_up = padder.unpad(flow_up).squeeze()
 
-            file_stem = imfile1.split('/')[-2]
+            file_stem = os.path.basename(imfile1).split('.')[0]
             if args.save_numpy:
                 np.save(output_directory / f"{file_stem}.npy", flow_up.cpu().numpy().squeeze())
             plt.imsave(output_directory / f"{file_stem}.png", -flow_up.cpu().numpy().squeeze(), cmap='jet')
