diff --git a/train_stereo.py b/train_stereo.py
index 29f82ed..b5d0062 100644
--- a/train_stereo.py
+++ b/train_stereo.py
@@ -150,7 +150,7 @@ def train(args):
     model.train()
     model.module.freeze_bn() # We keep BatchNorm frozen
 
-    validation_frequency = 10000
+    validation_frequency = 500
 
     scaler = GradScaler(enabled=args.mixed_precision)
 
@@ -185,9 +185,9 @@ def train(args):
                 logging.info(f"Saving file {save_path.absolute()}")
                 torch.save(model.state_dict(), save_path)
 
-                results = validate_things(model.module, iters=args.valid_iters)
+                # results = validate_things(model.module, iters=args.valid_iters)
 
-                logger.write_dict(results)
+                # logger.write_dict(results)
 
                 model.train()
                 model.module.freeze_bn()
@@ -256,4 +256,4 @@ if __name__ == '__main__':
 
     Path("checkpoints").mkdir(exist_ok=True, parents=True)
 
-    train(args)
\ No newline at end of file
+    train(args)
